util/test/box_testutil.go

```diff
@@ -0,0 +1,146 @@
+		}
+		accts[account] = acctData
+	}
+	return account, name, acctData
+}
+
+func addBoxInfoToStats(t *testing.T, fullKey string, value *string,
+	accts map[basics.Address]*ledgercore.AccountData, boxTotals map[basics.Address]basics.AccountData) {
+	addr, name, acctData := getNameAndAccountPointer(t, value, fullKey, accts)
+
+	acctData.TotalBoxes++
+	acctData.TotalBoxBytes += uint64(len(name) + len(*value))
+
+	boxTotals[addr] = basics.AccountData{
+		TotalBoxes:    acctData.TotalBoxes,
+		TotalBoxBytes: acctData.TotalBoxBytes,
+	}
+}
+
+func subtractBoxInfoToStats(t *testing.T, fullKey string, value *string,
+	accts map[basics.Address]*ledgercore.AccountData, boxTotals map[basics.Address]basics.AccountData) {
+	addr, name, acctData := getNameAndAccountPointer(t, value, fullKey, accts)
+
+	prevBoxBytes := uint64(len(name) + len(*value))
+	require.GreaterOrEqual(t, acctData.TotalBoxes, uint64(0))
+	require.GreaterOrEqual(t, acctData.TotalBoxBytes, prevBoxBytes)
+
+	acctData.TotalBoxes--
+	acctData.TotalBoxBytes -= prevBoxBytes
+
+	boxTotals[addr] = basics.AccountData{
+		TotalBoxes:    acctData.TotalBoxes,
+		TotalBoxBytes: acctData.TotalBoxBytes,
+	}
+}
+
+// BuildAccountDeltasFromKvsAndMods simulates keeping track of the evolution of the box statistics
+func BuildAccountDeltasFromKvsAndMods(t *testing.T, kvOriginals, kvMods map[string]*string) (
+	ledgercore.StateDelta, map[string]*string, map[basics.Address]basics.AccountData) {
+	kvUpdated := map[string]*string{}
+	boxTotals := map[basics.Address]basics.AccountData{}
+	accts := map[basics.Address]*ledgercore.AccountData{}
+	/*
+		1. fill the accts and kvUpdated using kvOriginals
+		2. for each (fullKey, value) in kvMod:
+			* (A) if the key is not present in kvOriginals just add the info as in #1
+			* (B) else (fullKey present):
+			    * (i)  if the value is nil
+					==> remove the box info from the stats and kvUpdated with assertions
+				* (ii) else (value is NOT nil):
+					==> reset kvUpdated and assert that the box hasn't changed shapes
+	*/
+
+	/* 1. */
+	for fullKey, value := range kvOriginals {
+		addBoxInfoToStats(t, fullKey, value, accts, boxTotals)
+		kvUpdated[fullKey] = value
+	}
+
+	/* 2. */
+	for fullKey, value := range kvMods {
+		prevValue, ok := kvOriginals[fullKey]
+		if !ok {
+			/* 2A. */
+			addBoxInfoToStats(t, fullKey, value, accts, boxTotals)
+			kvUpdated[fullKey] = value
+			continue
+		}
+		/* 2B. */
+		if value == nil {
+			/* 2Bi. */
+			subtractBoxInfoToStats(t, fullKey, prevValue, accts, boxTotals)
+			delete(kvUpdated, fullKey)
+			continue
+		}
+		/* 2Bii. */
+		require.Equal(t, len(*prevValue), len(*value))
+		require.Contains(t, kvUpdated, fullKey)
+		kvUpdated[fullKey] = value
+	}
+
+	var delta ledgercore.StateDelta
+	for acct, acctData := range accts {
+		delta.Accts.Upsert(acct, *acctData)
+	}
+	return delta, kvUpdated, boxTotals
+}
+
+// CompareAppBoxesAgainstLedger uses LedgerForEvaluator to assert that provided app boxes can be retrieved as expected
```

Comment:
Unused?

Reply:
used in [ledger_for_evaluator_test.go](https://github.com/algorand/indexer/blob/a17bc4b54e068c25204e0634ad9aea191a8d264b/processor/eval/ledger_for_evaluator_test.go#L618)

Reply:
In #1117 it's [moved into that file](https://github.com/algorand/indexer/blob/5500b7f4878ece92a67afa26944acf9abd73bd31/processor/eval/ledger_for_evaluator_test.go#L548)